src/Host/Monitor/Impl/MainWindow.xaml.cs

```diff
@@ -13,53 +16,52 @@ public partial class MainWindow : Window {
+                    if (await CredentialManager.IsBrokerUserCredentialSavedAsync()) {
+                        await BrokerManager.CreateOrAttachToBrokerInstanceAsync();
+                    } else {
+                        await CredentialManager.GetAndSaveCredentialsFromUserAsync();
+                    }
                 } else {
-                    await CredentialManager.GetAndSaveCredentialsFromUserAsync();
+                    await BrokerManager.CreateOrAttachToBrokerInstanceAsync();
                 }
-            } else {
-                await BrokerManager.StartBrokerInstanceAsync();
+            } catch (Exception ex) {
+                SetStatusText(ex.Message);
             }
         }
-
-        private async void StartBrokerBtn_Click(object sender, RoutedEventArgs e) {
+        private void StartBrokerBtn_Click(object sender, RoutedEventArgs e) {
             BrokerManager.ResetAutoStart();
-            if (Properties.Settings.Default.UseDifferentBrokerUser) {
-                if (await CredentialManager.IsBrokerUserCredentialSavedAsync()) {
-                    await BrokerManager.StartBrokerInstanceAsync();
-                } else {
-                    await CredentialManager.GetAndSaveCredentialsFromUserAsync();
-                }
-            } else {
-                await BrokerManager.StartBrokerInstanceAsync();
-            }
+            StartUpAsync().DoNotWait();
         }
-        private async void StopBrokerBtn_Click(object sender, RoutedEventArgs e) {
-            await BrokerManager.StopBrokerInstanceAsync();
+        private void StopBrokerBtn_Click(object sender, RoutedEventArgs e) {
+            BrokerManager.StopBrokerInstanceAsync().DoNotWait();
         }
 
         private async void AddOrChangeBrokerUserBtn_Click(object sender, RoutedEventArgs e) {
-            if(await CredentialManager.IsBrokerUserCredentialSavedAsync()) {
-                await CredentialManager.RemoveCredentialsAsync();
+            try {
+                if (await CredentialManager.IsBrokerUserCredentialSavedAsync()) {
+                    CredentialManager.RemoveCredentials();
+                }
+                await CredentialManager.GetAndSaveCredentialsFromUserAsync();
+            } catch (Exception ex) {
+                SetStatusText(ex.Message);
             }
-            await CredentialManager.GetAndSaveCredentialsFromUserAsync();
         }
-        private async void RemoveBrokerUserBtn_Click(object sender, RoutedEventArgs e) {
-            await CredentialManager.RemoveCredentialsAsync();
+        private void RemoveBrokerUserBtn_Click(object sender, RoutedEventArgs e) {
+            CredentialManager.RemoveCredentials();
         }
 
-        public static async void SetStatusText(string message) {
-            await _currentWindow.Dispatcher.InvokeAsync(() => {
+        public static void SetStatusText(string message) {
+            _currentWindow.Dispatcher.Invoke(() => {
```

Comment:
If `SetStatusText` is only ever called from event handlers and tasks that they spawn, I think this isn't necessary, since `await` will return to the UI context.